File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int l = 0, r = heights.size()-1, maxArea = 0;
        while (l < r) {
            maxArea = max(min(heights[l],heights[r])*(r-l),maxArea);
            if (heights[l] < heights[r]) {
                l++;
            } else {
                r--;
            }
        }
        return maxArea;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curr = nums[0]
        maxF = nums[0]
        for i in range(1,len(nums)):
            curr = max(nums[i], curr + nums[i])
            maxF = max(maxF, curr)
        return maxF